libs/hdf-converters/src/converters-from-hdf/asff/reverse-asff-mapper.ts

```diff
@@ -175,14 +185,151 @@ export class FromHdfToAsffMapper extends FromHdfBaseConverter {
+      ).length;
+      let size = originalSize;
+      let popped;
+      while (
+        size > SIZE_CAP &&
+        (finding.FindingProviderFields.Types as string[]).length > 0
+      ) {
+        popped = (finding.FindingProviderFields.Types as string[]).pop();
+        size = new TextEncoder().encode(JSON.stringify(finding)).length;
+      }
+      if (size > SIZE_CAP) {
+        console.error(
+          `Warning: Normalized entry could not be sufficiently reduced in size to meet AWS Security Hub requirements and so will not be provided in the results set.  Entry could not be minimized more than as follows:
+            ${finding}`
+        );
+        if (finding.Id === profileInfoFindingId) {
+          console.error(
+            'Warning: This entry was the informational one containing the scan/execution level information.'
+          );
+        }
+        if (finding.Id !== profileInfoFindingId) {
+          // metrics are only for non-profile info findings
+          numRemoved++;
+        }
+        continue;
+      }
+      if (originalSize !== size) {
+        (finding.FindingProviderFields.Types as string[]).push(
+          new TextDecoder().decode(
+            new TextEncoder().encode(popped).subarray(0, SIZE_CAP - size)
+          )
+        );
+        (finding.FindingProviderFields.Types as string[]).push(
+          'HDF2ASFF-converter/warning/Not all information was captured in this entry.  Please consult the original file for all of the information.'
+        );
+        console.error(
+          `Warning: Normalized entry was truncated in size to meet AWS Security Hub requirements.  Entry id: ${finding.Id}`
+        );
+        if (finding.Id !== profileInfoFindingId) {
+          numTruncated++;
+        }
+      }
+
+      // FindingProviderFields.Types has a maximum size of 50 attributes
+      const cutoff = (finding.FindingProviderFields.Types as string[]).splice(
+        50,
+        (finding.FindingProviderFields.Types as string[]).length - 50
+      );
+      if (cutoff.length > 0) {
+        (finding.FindingProviderFields.Types as string[]).pop();
+        (finding.FindingProviderFields.Types as string[]).push(
+          `HDF2ASFF-converter/warning/Not all information was captured in this entry.  Please consult the original file for all of the information.`
+        );
+        console.error(
+          `Warning: Normalized entry was truncated in size to meet AWS Security Hub requirements.  Entry id: ${finding.Id}`
+        );
+        if (finding.Id !== profileInfoFindingId) {
+          numTruncated++;
+        }
+      }
+
+      // sechub doesn't allow two types to have the same values; future iteration should find a way to work around this instead of just skipping it like we're going to do here (maybe add a number of {MAKE LINE DIFFERENT} block things at the end of an otherwise same line that'll get removed by the asff2hdf mapper similar to how we do the slash substitutions
```

Comment:
```suggestion
      // SecHub doesn't allow two types to have the same values; future iteration should find a way to work around this instead of just skipping it like we're going to do here (maybe add a number of {MAKE LINE DIFFERENT} block things at the end of an otherwise same line that'll get removed by the asff2hdf mapper similar to how we do the slash substitutions
```

Reply:
done